### PR TITLE
TC's refactoring and modification of tc_kernel_minimal defconfig

### PR DESCRIPTION
**Pull request by kishore-sn**
Changes:
	1) Remove pthread dependency on mqueue tc's.
	2) Replace pthread_create with task_create in some tc's.
	3) Add config checks such as CONFIG_DISABLE_SIGNALS, CONFIG_DISABLE_PTHREADS
	   etc, to prevent build errors.
	4) Modify tc_kernel_minimal defconfig to test the features supported by minimal config

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [a19b80cebd284e905b4d549b33c24b81ab983f92] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [a19b80cebd284e905b4d549b33c24b81ab983f92] - Code Rule Check OK.

File: apps/examples/testcase/le_tc/kernel/tc_mqueue.c [hunk 1]
```diff
@@ -561,106 +565,50 @@ static void *receiver_thread(void *arg)
  */
 static void tc_mqueue_mq_open_close_send_receive(void)
 {
-	pthread_t sender;
-	pthread_t receiver;
-	void *result;
-	pthread_attr_t attr;
-	struct sched_param sparam;
-	FAR void *expected;
+	const char* sendername = "Sender";
```
Comment (shadaksharikp) [inline apps/examples/testcase/le_tc/kernel/tc_mqueue.c]: Coding guidelines violated

File: apps/examples/testcase/le_tc/kernel/tc_sched.c [hunk 2]
```diff
@@ -44,7 +44,9 @@
 #define PID_IDLE        0
 #define TASK_CANCEL_INVALID  -1
 
+#ifndef CONFIG_DISABLE_PTHREAD
 pthread_t thread1, thread2;
```
Comment (shadaksharikp) [inline apps/examples/testcase/le_tc/kernel/tc_sched.c]: each variable should be on new line

**Comment by seinfra** [comment]
Target : [a19b80cebd284e905b4d549b33c24b81ab983f92] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [a19b80cebd284e905b4d549b33c24b81ab983f92] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [a19b80cebd284e905b4d549b33c24b81ab983f92] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [a19b80cebd284e905b4d549b33c24b81ab983f92] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [a19b80cebd284e905b4d549b33c24b81ab983f92] - Code Rule Check (C++) OK.

**Comment by sunghan-chang** [comment]
@kishore-sn This is unnecessary. Let's close.

**Pull request closed without merging**